app/src/composables/use-relation-multiple.ts

```diff
@@ -102,16 +102,18 @@ export function useRelationMultiple(
 			);
 			const deleteIndex = _value.value.delete.findIndex((id) => id === item[targetPKField]);
 
-			const updatedItem: Record<string, any> = {};
+			const updatedItem: Record<string, any> = cloneDeep(item);
 
 			if (editsIndex !== -1) {
-				merge(
+				mergeWith(
 					updatedItem,
 					{ $type: 'updated', $index: editsIndex, $edits: editsIndex },
-					_value.value.update[editsIndex]
+					_value.value.update[editsIndex],
+					(_objValue, srcValue) => {
+						// skip the default merge behavior for objects/arrays
+						if (typeof srcValue === 'object') return srcValue;
```

Comment:
Perhaps this should be something like lodash's `isObjectLike` to safer and avoid false positives on null values?

Reply:
Yeah good one. Effectively everything in JavaScript is an "object", so you might get a lot of false positives with a `typeof` check 

<img width="151" alt="CleanShot 2022-08-30 at 14 45 20@2x" src="https://user-images.githubusercontent.com/9141017/187517967-99462235-b177-4355-9942-026296def683.png">

Reply:
Although I believe the listed "false positives" will still pass `isObjectLike` (since the goal was to deny `null` from passing):

> Checks if value is object-like. A value is object-like if it's not null and has a typeof result of "object"

I think these "false positive" should technically still be correct since Date/Map/Set etc wouldn't do the magical object-append/array-concat, or even if they do, we'd want them to override as well right?